Applications/Spire/Source/Ui/CalendarDatePicker.cpp

```diff
@@ -0,0 +1,112 @@
+#include "Spire/Ui/CalendarDatePicker.hpp"
+#include <QVBoxLayout>
+#include "Spire/Spire/Dimensions.hpp"
+#include "Spire/Spire/LocalScalarValueModel.hpp"
+#include "Spire/Spire/ToTextModel.hpp"
+#include "Spire/Ui/Button.hpp"
+#include "Spire/Ui/CustomQtVariants.hpp"
+#include "Spire/Ui/TextBox.hpp"
+
+using namespace boost::gregorian;
+using namespace boost::signals2;
+using namespace Spire;
+using namespace Spire::Styles;
+
+class CalendarDatePicker::MonthSpinner : public QWidget {
+  public:
+    explicit MonthSpinner(QWidget* parent = nullptr)
+      : MonthSpinner(
+          std::make_shared<LocalDateModel>(day_clock::local_day()), parent) {}
```

Comment:
This should never be creating its own model, you can likely get rid of this constructor and the one on line 21.